src/stan/lang/ast/sigs/function_signatures_def.hpp

```diff
@@ -224,6 +224,32 @@ namespace stan {
           expr_type(double_type()));
     }
 
+    template<typename T>
+    expr_type function_signatures::rng_return_type(const expr_type& t) {
+      T return_type;
+      return t.is_primitive() ?
+        expr_type(return_type) : expr_type(return_type, 1U);
+    }
+
+    template<typename T>
+    expr_type function_signatures::rng_return_type(
+                            const expr_type& t,
+                            const expr_type& u) {
+      T return_type;
+      return t.is_primitive() && u.is_primitive()
+        ? expr_type(return_type) : expr_type(return_type, 1U);
+    }
+
+    template<typename T>
+    expr_type function_signatures::rng_return_type(
+                            const expr_type& t,
+                            const expr_type& u,
+                            const expr_type& v) {
+      T return_type;
+      return t.is_primitive() && (u.is_primitive() && v.is_primitive())
```

Comment:
This would be clearer if it's chained to just be

```
return rng_return_type(rng_return_type(t, u), v);
```

Though not relevant if you update, you don't need parens around `a && b && c`.  By default, it'll be left associative and resolve as `(a && b) && c`, which is fine.